ui-tests/src/test/java/org/openmrs/reference/FormTest.java

```diff
@@ -0,0 +1,117 @@
+import org.openmrs.reference.groups.BuildTests;
+import org.openmrs.reference.page.AdministrationPage;
+import org.openmrs.reference.page.ClinicianFacingPatientDashboardPage;
+import org.openmrs.reference.page.HeaderPage;
+import org.openmrs.reference.page.HtmlFormsPage;
+import org.openmrs.reference.page.ManageHtmlFormsPage;
+import org.openmrs.reference.page.HomePage;
+import org.openmrs.reference.page.ManageFormsPage;
+import org.openqa.selenium.By;
+import org.openmrs.uitestframework.test.TestBase;
+
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Created by nata on 24.06.15.
+ */
+public class FormTest extends TestBase {
+	
+	private static String name = "newFormTest1";
+	
+	private static String description = "description of new form";
+	
+	private static String version = "1.2";
+	
+	private HomePage homePage;
+	
+	private AdministrationPage administrationPage;
+	
+	private HeaderPage headerPage;
+	
+	private ManageFormsPage manageForm;
+	
+	private ManageHtmlFormsPage manageHtmlFormsPage;
+	
+	private HtmlFormsPage htmlFormsPage;
+	
+	private ClinicianFacingPatientDashboardPage patientDashboardPage;
+	
+	@Before
+	public void setUp() throws Exception {
+		homePage = new HomePage(page);
+		assertPage(homePage.waitForPage());
+		headerPage = new HeaderPage(driver);
+		administrationPage = new AdministrationPage(page);
+		manageForm = new ManageFormsPage(driver);
+		htmlFormsPage = new HtmlFormsPage(page);
+		manageHtmlFormsPage = new ManageHtmlFormsPage(page);
+		patientDashboardPage = new ClinicianFacingPatientDashboardPage(page);
+	}
+	
+	private void initiateFormPresence() throws InterruptedException {
+		manageHtmlFormsPage = homePage.goToAdministration().clickOnManageHtmlForms();
+		if (manageHtmlFormsPage.getElementsIfExisting(By.xpath("//*[contains(text(), '" + name + "')]")).isEmpty()) {
+			manageHtmlFormsPage.clickOnNewHtmlForm();
+			htmlFormsPage.createNewFormTest(name, description, version);
+		}
+		homePage.go();
+		homePage.goToManageForm();
+		if (!manageForm.addPresent()) {
+			manageForm.delete();
+		}
+	}
+	
+	@Test
+	@Category(BuildTests.class)
+	public void addFormTest() throws Exception {
+		
+		initiateFormPresence();
+		manageForm.add();
+		manageForm.addLabel("Eye Report");
+		manageForm.addIcon("icon-align-justify");
+		manageForm.formIdFromUrl();
+		manageForm.save();
+		headerPage.clickOnHomeIcon();
+		assertNotNull("Eye Report", patientDashboardPage.FORM_EXIST);
+	}
+	
+	@Test
+	@Category(BuildTests.class)
+	public void editFormTest() throws Exception {
+		
+		addFormTest();
+		homePage.goToManageForm();
+		manageForm.waitForPage();
+		manageForm.editPath();
+		manageForm.addLabel("Eye Test");
+		manageForm.save();
+		headerPage.clickOnHomeIcon();
+		assertNotNull("Eye Test", patientDashboardPage.FORM_EXIST);
+	}
+	
+	@Test
+	@Category(BuildTests.class)
+	public void deleteFormTest() throws Exception {
+		
+		addFormTest();
+		headerPage.clickOnHomeIcon();
```

Comment:
Shouldn't we already have done this by calling `addFormTest()`?

Reply:
We decided to merge all the three functionality, AddForm, editForm And DeleteForm simply because we needed to have test independent,because if we had separated them , each test would be dependant which would not be appropriate to handle while running each specific test

Reply:
I get that and it makes sense. It's just that the last active line of `addFormTest()` is [`headerPage.clickOnHomeIcon()`](https://github.com/openmrs/openmrs-distro-referenceapplication/pull/503/files/da2b612b843e37731990ac581354c07e4a2a92f6#diff-40f58e09e158a43ef8df5dc6de3ea2f206cca3b6dc09552fd200ddea0ef795bbR80), so, in other words, isn't this test already starting with this having been done?

Reply:
ohh yeah true , Let me refactor that